lib/sensors/src/URM04Sensor.cpp

```diff
@@ -0,0 +1,131 @@
+#include "URM04Sensor.h"
+
+// instantiate pin connected to the URM04 URM04Sensor
+URM04Sensor::URM04Sensor::URM04Sensor(PinName trig_pin, PinName _RX, PinName _TX)
+    : m_trigPin(trig_pin), startAddr(START_ADDRESS), RX(_RX), TX(_TX), serial(TX, RX, BAUD_RATE) {
```

Comment:
Since we have 4 ultrasonic sensors, we need the option of setting a different address to a sensor other than the default address. Create a function that executes the set device address command.

Reply:
alright just made another commit

Reply:
Alright, I finished implementing it. My only question is any time I change from reading to writing to the serial within the function do I have to clear the buffer `serial.sync()`.